Build failure: streamripper

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

nix build .#streamripper -L

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/322496537

### Relevant log output

```console
streamripper> Running phase: unpackPhase
streamripper> unpacking source archive /nix/store/9p7c7z53j0jxq7rj5p8fjmbpn8bxlg84-streamripper-1.64.6.tar.gz
streamripper> source root is streamripper-1.64.6
streamripper> setting SOURCE_DATE_EPOCH to timestamp 1238464302 of file "streamripper-1.64.6/lib/config.h.in"
streamripper> Running phase: patchPhase
streamripper> applying patch /nix/store/g8n06r8jvwyb09klksglapqk3q52vd0c-1075541-gcc14
streamripper> patching file lib/argv.c
streamripper> applying patch /nix/store/vqmsv4pnxzs8zylg55i4rrz9hz451qkn-873964-http
streamripper> patching file lib/http.c
streamripper> patching file lib/http.h
streamripper> applying patch /nix/store/p2fn0paiyricbn3my6bln547zfbvrj9w-streamripper-http-1.0.patch
streamripper> patching file lib/http.c
streamripper> Hunk #1 succeeded at 238 (offset 10 lines).
streamripper> Running phase: updateAutotoolsGnuConfigScriptsPhase
streamripper> Updating Autotools / GNU config script to a newer upstream version: ./config.sub
streamripper> Updating Autotools / GNU config script to a newer upstream version: ./libmad-0.15.1b/config.sub
streamripper> Updating Autotools / GNU config script to a newer upstream version: ./config.guess
streamripper> Updating Autotools / GNU config script to a newer upstream version: ./libmad-0.15.1b/config.guess
streamripper> Running phase: configurePhase
streamripper> fixing libtool script ./libmad-0.15.1b/ltmain.sh
streamripper> fixing libtool script ./ltmain.sh
streamripper> patching script interpreter paths in ./configure
streamripper> ./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/2hjsch59amjs3nbgh7ahcfzm2bfwl8zi-bash-5.3p9/bin/sh"
streamripper> configure flags: --disable-dependency-tracking --prefix=/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6
streamripper> checking for a BSD-compatible install... /nix/store/hlxw2q9qansq7bn52xvlb5badw3z1v8s-coreutils-9.10/bin/install -c
streamripper> checking whether build environment is sane... yes
streamripper> checking for a thread-safe mkdir -p... /nix/store/hlxw2q9qansq7bn52xvlb5badw3z1v8s-coreutils-9.10/bin/mkdir -p
streamripper> checking for gawk... gawk
streamripper> checking whether make sets $(MAKE)... yes
streamripper> checking for gcc... gcc
streamripper> checking for C compiler default output file name... a.out
streamripper> checking whether the C compiler works... yes
streamripper> checking whether we are cross compiling... no
streamripper> checking for suffix of executables...
streamripper> checking for suffix of object files... o
streamripper> checking whether we are using the GNU C compiler... yes
streamripper> checking whether gcc accepts -g... yes
streamripper> checking for gcc option to accept ISO C89... none needed
streamripper> checking for style of include used by make... GNU
streamripper> checking dependency style of gcc... none
streamripper> checking for a BSD-compatible install... /nix/store/hlxw2q9qansq7bn52xvlb5badw3z1v8s-coreutils-9.10/bin/install -c
streamripper> checking for ranlib... ranlib
streamripper> checking for inline... inline
streamripper> checking for main in -lm... yes
streamripper> checking how to run the C preprocessor... gcc -E
streamripper> checking for grep that handles long lines and -e... /nix/store/8laf6k81j9ckylrigj3xsk76j69knhvl-gnugrep-3.12/bin/grep
streamripper> checking for egrep... /nix/store/8laf6k81j9ckylrigj3xsk76j69knhvl-gnugrep-3.12/bin/grep -E
streamripper> checking for ANSI C header files... yes
streamripper> checking for sys/types.h... yes
streamripper> checking for sys/stat.h... yes
streamripper> checking for stdlib.h... yes
streamripper> checking for string.h... yes
streamripper> checking for memory.h... yes
streamripper> checking for strings.h... yes
streamripper> checking for inttypes.h... yes
streamripper> checking for stdint.h... yes
streamripper> checking for unistd.h... yes
streamripper> checking for working alloca.h... yes
streamripper> checking for alloca... yes
streamripper> checking pkg-config is at least version 0.7... yes
streamripper> checking for GLIB - version >= 2.16.0... yes (version 2.86.3)
streamripper> checking for mad_stream_buffer in -lmad... yes
streamripper> checking for pthread_create in -lpthread... yes
streamripper> checking for socklen_t... yes
streamripper> checking for dirent.h that defines DIR... yes
streamripper> checking for library containing opendir... none required
streamripper> checking for ANSI C header files... (cached) yes
streamripper> checking for unistd.h... (cached) yes
streamripper> checking wchar.h usability... yes
streamripper> checking wchar.h presence... yes
streamripper> checking for wchar.h... yes
streamripper> checking wctype.h usability... yes
streamripper> checking wctype.h presence... yes
streamripper> checking for wctype.h... yes
streamripper> checking for wchar_t... yes
streamripper> checking for iswcntrl... yes
streamripper> checking for vswprintf... yes
streamripper> checking whether mbrtowc and mbstate_t are properly declared... yes
streamripper> checking build system type... x86_64-pc-linux-gnu
streamripper> checking host system type... x86_64-pc-linux-gnu
streamripper> checking for ld used by GCC... ld
streamripper> checking if the linker (ld) is GNU ld... yes
streamripper> checking for shared library run path origin... done
streamripper> checking for iconv... yes
streamripper> checking for iconv declaration...
streamripper>          extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
streamripper> checking for nl_langinfo and CODESET... yes
streamripper> checking for buildargv in -liberty... no
streamripper> checking for NeAACDecDecode2 in -lfaad... no
streamripper> checking for initscr in -lncurses ... no
streamripper> checking for initscr in -lcurses  ... no
streamripper> checking for initscr in -lncursesw... no
streamripper> checking for initscr in -lpdcurses... no
streamripper> configure: Request no curses
streamripper> checking for Ogg... yes
streamripper> checking for Vorbis... yes
streamripper> checking for sem_init... yes
streamripper> checking for sem_init... (cached) yes
streamripper> checking for stdint.h... (cached) yes
streamripper> checking for inttypes.h... (cached) yes
streamripper> checking for uint32_t... yes
streamripper> checking for u_int32_t... yes
streamripper> checking sys/sockio.h usability... no
streamripper> checking sys/sockio.h presence... no
streamripper> checking for sys/sockio.h... no
streamripper> checking for connect... yes
streamripper> checking for size_t... yes
streamripper> checking return type of signal handlers... void
streamripper> checking for vprintf... yes
streamripper> checking for _doprnt... no
streamripper> checking for mkdir... yes
streamripper> checking for socket... yes
streamripper> checking for strerror... yes
streamripper> checking for strstr... yes
streamripper> configure: creating ./config.status
streamripper> config.status: creating lib/Makefile
streamripper> config.status: creating Makefile
streamripper> config.status: creating lib/config.h
streamripper> config.status: executing depfiles commands
streamripper> * Streamripper configuration
streamripper>     Using included MAD library?      no
streamripper>     Ogg/vorbis support?              yes
streamripper> Running phase: buildPhase
streamripper> build flags: SHELL=/nix/store/2hjsch59amjs3nbgh7ahcfzm2bfwl8zi-bash-5.3p9/bin/bash
streamripper> Making all in lib
streamripper> make[1]: Entering directory '/build/streamripper-1.64.6/lib'
streamripper> make  all-am
streamripper> make[2]: Entering directory '/build/streamripper-1.64.6/lib'
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c cbuf2.c
streamripper> In file included from srtypes.h:25,
streamripper>                  from errors.h:4,
streamripper>                  from cbuf2.c:43:
streamripper> cbuf2.c: In function 'cbuf2_advance_metadata_list':
streamripper> list.h:211:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   211 |         for (pos = (head)->prev, prefetch(pos->prev); pos != (head); \
streamripper>       |                                ^
streamripper> cbuf2.c:140:5: note: in expansion of macro 'list_for_each_prev'
streamripper>   140 |     list_for_each_prev (p, &(cbuf2->metadata_list)) {
streamripper>       |     ^~~~~~~~~~~~~~~~~~
streamripper> list.h:212:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   212 |                 pos = pos->prev, prefetch(pos->prev))
streamripper>       |                                ^
streamripper> cbuf2.c:140:5: note: in expansion of macro 'list_for_each_prev'
streamripper>   140 |     list_for_each_prev (p, &(cbuf2->metadata_list)) {
streamripper>       |     ^~~~~~~~~~~~~~~~~~
streamripper> cbuf2.c: In function 'cbuf2_insert_chunk':
streamripper> list.h:203:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   203 |         for (pos = (head)->next, prefetch(pos->next); pos != (head); \
streamripper>       |                                ^
streamripper> cbuf2.c:378:9: note: in expansion of macro 'list_for_each'
streamripper>   378 |         list_for_each (p, &this_page_list) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> list.h:204:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   204 |                 pos = pos->next, prefetch(pos->next))
streamripper>       |                                ^
streamripper> cbuf2.c:378:9: note: in expansion of macro 'list_for_each'
streamripper>   378 |         list_for_each (p, &this_page_list) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> cbuf2.c: In function 'cbuf2_extract_relay_mp3':
streamripper> list.h:203:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   203 |         for (pos = (head)->next, prefetch(pos->next); pos != (head); \
streamripper>       |                                ^
streamripper> cbuf2.c:521:9: note: in expansion of macro 'list_for_each'
streamripper>   521 |         list_for_each (p, &(cbuf2->metadata_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> list.h:204:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   204 |                 pos = pos->next, prefetch(pos->next))
streamripper>       |                                ^
streamripper> cbuf2.c:521:9: note: in expansion of macro 'list_for_each'
streamripper>   521 |         list_for_each (p, &(cbuf2->metadata_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> cbuf2.c: In function 'cbuf2_debug_lists':
streamripper> list.h:203:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   203 |         for (pos = (head)->next, prefetch(pos->next); pos != (head); \
streamripper>       |                                ^
streamripper> cbuf2.c:555:9: note: in expansion of macro 'list_for_each'
streamripper>   555 |         list_for_each (p, &(cbuf2->metadata_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> list.h:204:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   204 |                 pos = pos->next, prefetch(pos->next))
streamripper>       |                                ^
streamripper> cbuf2.c:555:9: note: in expansion of macro 'list_for_each'
streamripper>   555 |         list_for_each (p, &(cbuf2->metadata_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> list.h:203:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   203 |         for (pos = (head)->next, prefetch(pos->next); pos != (head); \
streamripper>       |                                ^
streamripper> cbuf2.c:572:9: note: in expansion of macro 'list_for_each'
streamripper>   572 |         list_for_each (p, &(cbuf2->ogg_page_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> list.h:204:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   204 |                 pos = pos->next, prefetch(pos->next))
streamripper>       |                                ^
streamripper> cbuf2.c:572:9: note: in expansion of macro 'list_for_each'
streamripper>   572 |         list_for_each (p, &(cbuf2->ogg_page_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> cbuf2.c: In function 'cbuf2_init_relay_entry':
streamripper> list.h:203:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   203 |         for (pos = (head)->next, prefetch(pos->next); pos != (head); \
streamripper>       |                                ^
streamripper> cbuf2.c:782:9: note: in expansion of macro 'list_for_each'
streamripper>   782 |         list_for_each (p, &(cbuf2->ogg_page_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> list.h:204:32: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   204 |                 pos = pos->next, prefetch(pos->next))
streamripper>       |                                ^
streamripper> cbuf2.c:782:9: note: in expansion of macro 'list_for_each'
streamripper>   782 |         list_for_each (p, &(cbuf2->ogg_page_list)) {
streamripper>       |         ^~~~~~~~~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c charset.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c debug.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c errors.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c external.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c filelib.c
streamripper> filelib.c: In function 'parse_and_subst_pat':
streamripper> filelib.c:805:9: warning: variable 'done' set but not used [-Wunused-but-set-variable]
streamripper>   805 |     int done;
streamripper>       |         ^~~~
streamripper> filelib.c: In function 'get_next_sequence_number':
streamripper> filelib.c:1338:9: warning: variable 'rc' set but not used [-Wunused-but-set-variable]
streamripper>  1338 |     int rc;
streamripper>       |         ^~
streamripper> filelib.c: In function 'filelib_adjust_cuefile':
streamripper> filelib.c:1215:5: warning: ignoring return value of 'fgets' declared with attribute 'warn_unused_result' [-Wunused-result]
streamripper>  1215 |     fgets (cue_buf, 1024, fp_in);
streamripper>       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c findsep.c
streamripper> findsep.c: In function 'findsep_silence':
streamripper> findsep.c:133:9: warning: variable 'result' set but not used [-Wunused-but-set-variable]
streamripper>   133 |     int result;
streamripper>       |         ^~~~~~
streamripper> In file included from srtypes.h:25,
streamripper>                  from findsep.h:21,
streamripper>                  from findsep.c:30:
streamripper> findsep.c: In function 'apply_padding':
streamripper> list.h:240:58: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   240 |         for (pos = list_entry((head)->next, type, member),      \
streamripper>       |                                                          ^
streamripper> findsep.c:251:5: note: in expansion of macro 'list_for_each_entry'
streamripper>   251 |     list_for_each_entry (pos, FRAME_LIST, &(ds->frame_list), m_list) {
streamripper>       |     ^~~~~~~~~~~~~~~~~~~
streamripper> list.h:243:62: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   243 |              pos = list_entry(pos->member.next, type, member),  \
streamripper>       |                                                              ^
streamripper> findsep.c:251:5: note: in expansion of macro 'list_for_each_entry'
streamripper>   251 |     list_for_each_entry (pos, FRAME_LIST, &(ds->frame_list), m_list) {
streamripper>       |     ^~~~~~~~~~~~~~~~~~~
streamripper> findsep.c: In function 'find_bitrate':
streamripper> findsep.c:488:9: warning: variable 'result' set but not used [-Wunused-but-set-variable]
streamripper>   488 |     int result;
streamripper>       |         ^~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c findsep2.c
streamripper> findsep2.c: In function 'findsep_silence_2':
streamripper> findsep2.c:130:9: warning: variable 'result' set but not used [-Wunused-but-set-variable]
streamripper>   130 |     int result;
streamripper>       |         ^~~~~~
streamripper> In file included from srtypes.h:25,
streamripper>                  from findsep.h:21,
streamripper>                  from findsep2.c:30:
streamripper> findsep2.c: In function 'apply_padding':
streamripper> list.h:240:58: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   240 |         for (pos = list_entry((head)->next, type, member),      \
streamripper>       |                                                          ^
streamripper> findsep2.c:250:5: note: in expansion of macro 'list_for_each_entry'
streamripper>   250 |     list_for_each_entry (pos, FRAME_LIST, &(ds->frame_list), m_list) {
streamripper>       |     ^~~~~~~~~~~~~~~~~~~
streamripper> list.h:243:62: warning: right-hand operand of comma expression has no effect [-Wunused-value]
streamripper>   243 |              pos = list_entry(pos->member.next, type, member),  \
streamripper>       |                                                              ^
streamripper> findsep2.c:250:5: note: in expansion of macro 'list_for_each_entry'
streamripper>   250 |     list_for_each_entry (pos, FRAME_LIST, &(ds->frame_list), m_list) {
streamripper>       |     ^~~~~~~~~~~~~~~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c http.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c iconvert.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c mchar.c
streamripper> mchar.c: In function 'set_codesets_default':
streamripper> mchar.c:575:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   575 |         strncpy (cs_opt->codeset_locale, fromcode, MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> mchar.c:576:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   576 |         strncpy (cs_opt->codeset_filesys, fromcode, MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> mchar.c:577:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   577 |         strncpy (cs_opt->codeset_metadata, fromcode, MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> mchar.c:578:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   578 |         strncpy (cs_opt->codeset_relay, fromcode, MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c parse.c
streamripper> parse.c: In function 'parse_metadata':
streamripper> parse.c:451:9: warning: variable 'eflags' set but not used [-Wunused-but-set-variable]
streamripper>   451 |     int eflags;
streamripper>       |         ^~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c prefs.c
streamripper> prefs.c: In function 'prefs_load':
streamripper> prefs.c:184:22: warning: 'padding' may be used uninitialized [-Wmaybe-uninitialized]
streamripper>   184 |             if (!rc1 || padding == 300) {
streamripper>       |                 ~~~~~^~~~~~~~~~~~~~~~~
streamripper> prefs.c:132:13: note: 'padding' was declared here
streamripper>   132 |         int padding;
streamripper>       |             ^~~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c relaylib.c
streamripper> relaylib.c: In function 'thread_send':
streamripper> relaylib.c:670:16: warning: variable 'err' set but not used [-Wunused-but-set-variable]
streamripper>   670 |     error_code err = SR_SUCCESS;
streamripper>       |                ^~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c rip_manager.c
streamripper> rip_manager.c: In function 'rip_manager_stop':
streamripper> rip_manager.c:152:5: warning: ignoring return value of 'write' declared with attribute 'warn_unused_result' [-Wunused-result]
streamripper>   152 |     write (rmi->abort_pipe[1], "0", 1);
streamripper>       |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c ripaac.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c ripogg.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c ripstream.c
streamripper> ripstream.c: In function 'ripstream_rip_ogg':
streamripper> ripstream.c:228:12: warning: variable 'extract_size' set but not used [-Wunused-but-set-variable]
streamripper>   228 |     u_long extract_size;
streamripper>       |            ^~~~~~~~~~~~
streamripper> ripstream.c: In function 'end_track_mp3':
streamripper> ripstream.c:676:9: warning: '__builtin_strncpy' output truncated before terminating nul copying 3 bytes from a string of the same length [-Wstringop-truncation]
streamripper>   676 |         strncpy (id3v1.tag, "TAG", strlen("TAG"));
streamripper>       |         ^
streamripper> ripstream.c: In function 'write_id3v2_frame.isra':
streamripper> ripstream.c:723:5: warning: '__builtin_strncpy' specified bound 4 equals destination size [-Wstringop-truncation]
streamripper>   723 |     strncpy(id3v2frame.id, tag_name, 4);
streamripper>       |     ^
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c socklib.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c threadlib.c
streamripper> gcc -DHAVE_CONFIG_H -I.  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c utf8.c
streamripper> utf8.c: In function 'utf8_decode':
streamripper> utf8.c:310:11: warning: unused variable 'charset' [-Wunused-variable]
streamripper>   310 |     char *charset;
streamripper>       |           ^~~~~~~
streamripper> rm -f libstreamripper.a
streamripper> ar cru libstreamripper.a cbuf2.o charset.o debug.o errors.o external.o filelib.o findsep.o findsep2.o http.o iconvert.o mchar.o parse.o prefs.o relaylib.o rip_manager.o ripaac.o ripogg.o ripstream.o socklib.o threadlib.o utf8.o
streamripper> ar: `u' modifier ignored since `D' is the default (see `U')
streamripper> ranlib libstreamripper.a
streamripper> make[2]: Leaving directory '/build/streamripper-1.64.6/lib'
streamripper> make[1]: Leaving directory '/build/streamripper-1.64.6/lib'
streamripper> make[1]: Entering directory '/build/streamripper-1.64.6'
streamripper> gcc -DHAVE_CONFIG_H -I. -I./lib  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c -o streamripper.o `test -f 'console/streamripper.c' || echo './'`console/streamripper.c
streamripper> console/streamripper.c: In function 'main':
streamripper> console/streamripper.c:343:5: warning: '__builtin_strncpy' specified bound 8192 equals destination size [-Wstringop-truncation]
streamripper>   343 |     strncpy (prefs->url, argv[1], MAX_URL_LEN);
streamripper>       |     ^
streamripper> console/streamripper.c:493:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   493 |             strncpy(prefs->rules_file, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:486:13: warning: '__builtin_strncpy' specified bound 1024 equals destination size [-Wstringop-truncation]
streamripper>   486 |             strncpy(prefs->useragent, argv[i], MAX_USERAGENT_STR);
streamripper>       |             ^
streamripper> console/streamripper.c:467:17: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   467 |                 strncpy(prefs->relay_ip, argv[i], SR_MAX_PATH);
streamripper>       |                 ^
streamripper> console/streamripper.c:442:13: warning: '__builtin_strncpy' specified bound 8192 equals destination size [-Wstringop-truncation]
streamripper>   442 |             strncpy(prefs->proxyurl, argv[i], MAX_URL_LEN);
streamripper>       |             ^
streamripper> console/streamripper.c:381:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   381 |             strncpy(prefs->output_directory, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:371:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   371 |             strncpy (prefs->showfile_pattern, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:421:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   421 |             strncpy(prefs->pls_file, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:408:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   408 |             strncpy(prefs->if_name, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:390:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   390 |             strncpy(prefs->ext_cmd, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:385:13: warning: '__builtin_strncpy' specified bound 254 equals destination size [-Wstringop-truncation]
streamripper>   385 |             strncpy(prefs->output_pattern, argv[i], SR_MAX_PATH);
streamripper>       |             ^
streamripper> console/streamripper.c:627:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   627 |         strncpy (prefs->cs_opt.codeset_relay, &rule[x], MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> console/streamripper.c:620:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   620 |         strncpy (prefs->cs_opt.codeset_metadata, &rule[x], MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> console/streamripper.c:613:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   613 |         strncpy (prefs->cs_opt.codeset_id3, &rule[x], MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> console/streamripper.c:606:9: warning: '__builtin_strncpy' specified bound 128 equals destination size [-Wstringop-truncation]
streamripper>   606 |         strncpy (prefs->cs_opt.codeset_filesys, &rule[x], MAX_CODESET_STRING);
streamripper>       |         ^
streamripper> gcc -DHAVE_CONFIG_H -I. -I./lib  -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/mwv93b79mh8fkzyimzk8l0a257d79q77-streamripper-1.64.6/include -I/nix/store/jhb15l57qmqxck5ddd1dh2ys729vc8gb-glib-2.86.3-dev/include/glib-2.0 -I/nix/store/1qm74vf93ik1xjrr9kl6qvjrklljlcqh-glib-2.86.3/lib/glib-2.0/include  -I/build/streamripper-1.64.6/lib -D__UNIX__  -g -O2 -Wall -c -o argv.o `test -f 'lib/argv.c' || echo './'`lib/argv.c
streamripper> lib/argv.c:65:14: warning: conflicting types for built-in function 'malloc'; expected 'void *(long unsigned int)' [-Wbuiltin-declaration-mismatch]
streamripper>    65 | extern void *malloc ();         /* Standard memory allocater */
streamripper>       |              ^~~~~~
streamripper> lib/argv.c:35:1: note: 'malloc' is declared in header '<stdlib.h>'
streamripper>    34 | # include <alloca.h>
streamripper>   +++ |+#include <stdlib.h>
streamripper>    35 | #elif defined __GNUC__
streamripper> lib/argv.c:66:14: warning: conflicting types for built-in function 'realloc'; expected 'void *(void *, long unsigned int)' [-Wbuiltin-declaration-mismatch]
streamripper>    66 | extern void *realloc ();        /* Standard memory reallocator */
streamripper>       |              ^~~~~~~
streamripper> lib/argv.c:66:14: note: 'realloc' is declared in header '<stdlib.h>'
streamripper> lib/argv.c:67:13: warning: conflicting types for built-in function 'free'; expected 'void(void *)' [-Wbuiltin-declaration-mismatch]
streamripper>    67 | extern void free ();            /* Free malloc'd memory */
streamripper>       |             ^~~~
streamripper> lib/argv.c:67:13: note: 'free' is declared in header '<stdlib.h>'
streamripper> lib/argv.c:68:14: error: conflicting types for 'strdup'; have 'char *(void)'
streamripper>    68 | extern char *strdup ();         /* Duplicate a string */
streamripper>       |              ^~~~~~
streamripper> In file included from lib/argv.c:33:
streamripper> /nix/store/j8irrc0mpx029dw0rmadsjylg7h31ync-glibc-2.42-51-dev/include/string.h:187:14: note: previous declaration of 'strdup' with type 'char *(const char *)'
streamripper>   187 | extern char *strdup (const char *__s)
streamripper>       |              ^~~~~~
streamripper> lib/argv.c: In function 'dupargv':
streamripper> lib/argv.c:100:1: warning: old-style function definition [-Wold-style-definition]
streamripper>   100 | dupargv (argv)
streamripper>       | ^~~~~~~
streamripper> lib/argv.c:111:20: error: too many arguments to function 'malloc'; expected 0, have 1
streamripper>   111 |   copy = (char **) malloc ((argc + 1) * sizeof (char *));
streamripper>       |                    ^~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
streamripper> lib/argv.c:65:14: note: declared here
streamripper>    65 | extern void *malloc ();         /* Standard memory allocater */
streamripper>       |              ^~~~~~
streamripper> lib/argv.c:119:20: error: too many arguments to function 'malloc'; expected 0, have 1
streamripper>   119 |       copy[argc] = malloc (sizeof (char *) * (len + 1));
streamripper>       |                    ^~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
streamripper> lib/argv.c:65:14: note: declared here
streamripper>    65 | extern void *malloc ();         /* Standard memory allocater */
streamripper>       |              ^~~~~~
streamripper> lib/argv.c: In function 'freeargv':
streamripper> lib/argv.c:144:6: warning: old-style function definition [-Wold-style-definition]
streamripper>   144 | void freeargv (vector)
streamripper>       |      ^~~~~~~~
streamripper> lib/argv.c:153:11: error: too many arguments to function 'free'; expected 0, have 1
streamripper>   153 |           free (*scan);
streamripper>       |           ^~~~  ~~~~~
streamripper> lib/argv.c:67:13: note: declared here
streamripper>    67 | extern void free ();            /* Free malloc'd memory */
streamripper>       |             ^~~~
streamripper> lib/argv.c:155:7: error: too many arguments to function 'free'; expected 0, have 1
streamripper>   155 |       free (vector);
streamripper>       |       ^~~~  ~~~~~~
streamripper> lib/argv.c:67:13: note: declared here
streamripper>    67 | extern void free ();            /* Free malloc'd memory */
streamripper>       |             ^~~~
streamripper> lib/argv.c: In function 'buildargv':
streamripper> lib/argv.c:199:8: warning: old-style function definition [-Wold-style-definition]
streamripper>   199 | char **buildargv (input)
streamripper>       |        ^~~~~~~~~
streamripper> lib/argv.c:230:37: error: too many arguments to function 'malloc'; expected 0, have 1
streamripper>   230 |                   nargv = (char **) malloc (maxargc * sizeof (char *));
streamripper>       |                                     ^~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~
streamripper> lib/argv.c:65:14: note: declared here
streamripper>    65 | extern void *malloc ();         /* Standard memory allocater */
streamripper>       |              ^~~~~~
streamripper> lib/argv.c:235:37: error: too many arguments to function 'realloc'; expected 0, have 2
streamripper>   235 |                   nargv = (char **) realloc (argv, maxargc * sizeof (char *));
streamripper>       |                                     ^~~~~~~  ~~~~
streamripper> lib/argv.c:66:14: note: declared here
streamripper>    66 | extern void *realloc ();        /* Standard memory reallocator */
streamripper>       |              ^~~~~~~
streamripper> lib/argv.c:309:24: error: too many arguments to function 'strdup'; expected 0, have 1
streamripper>   309 |           argv[argc] = strdup (copybuf);
streamripper>       |                        ^~~~~~  ~~~~~~~
streamripper> lib/argv.c:68:14: note: declared here
streamripper>    68 | extern char *strdup ();         /* Duplicate a string */
streamripper>       |              ^~~~~~
streamripper> make[1]: *** [Makefile:750: argv.o] Error 1
streamripper> make[1]: Leaving directory '/build/streamripper-1.64.6'
streamripper> make: *** [Makefile:811: all-recursive] Error 1
error: Cannot build '/nix/store/9imhdgm1dglwww03qcgabsap6c8gmj7f-streamripper-1.64.6.drv'.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.19.2, NixOS, 26.05 (Yarara), 26.05pre948083.0182a3613243`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.33.3`
 - nixpkgs: `/home/vladimir/work/extern/nixpkgs`


### Notify maintainers

@mdaniels5757 @Aleksanaa (following https://github.com/NixOS/nixpkgs/pull/373144 / https://github.com/NixOS/nixpkgs/pull/415992)

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/st/streamripper/package.nix
+++ b/pkgs/by-name/st/streamripper/package.nix
@@ -35,6 +35,13 @@ stdenv.mkDerivation rec {
       patch = "873964-http";
       hash = "sha256-D6koUCbnJHtRuq2zZy9VrxymuGXN1COacbQhphgB8qo=";
     })
+    # fix build with gcc 15
+    (fetchDebianPatch {
+      inherit pname version;
+      debianRevision = "4";
+      patch = "1097944-gcc15";
+      hash = "sha256-yBFDxd2sNlavQDmg/MCORFdpJY8p1Lzo131T4sBby5g=";
+    })
     # fix SR_ERROR_INVALID_METADATA caused by HTTP chunking
     # (https://sourceforge.net/p/streamripper/bugs/193/#6a82)
     (fetchpatch {

--- a/pkgs/by-name/st/streamripper/package.nix
+++ b/pkgs/by-name/st/streamripper/package.nix
@@ -64,5 +64,6 @@ stdenv.mkDerivation rec {
     description = "Application that lets you record streaming mp3 to your hard drive";
     license = lib.licenses.gpl2;
     mainProgram = "streamripper";
+    maintainers = with lib.maintainers; [ cybershadow ];
   };
 }

